cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -269,98 +270,99 @@ private KeyStore setupTrustStore(X509Certificate caCertKafka, char[] password, X
-                        Base64.Decoder decoder = Base64.getDecoder();
-                        char[] password = "password".toCharArray();
-
-                        // hack
-                        Pod pod = podOperations.get(namespace, name + "-" + i);
-                        SecretOperator secretOperations = new SecretOperator(vertx, client);
-                        String cluster = pod.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
-                        Secret brokersSecret = secretOperations.get(namespace, KafkaCluster.brokersSecretName(cluster));
-                        Secret clusterCaKeySecret = secretOperations.get(namespace, cluster + "-cluster-ca");
-                        Secret clusterCaCertificateSecret = secretOperations.get(namespace, cluster + "-cluster-ca-cert");
-                        CertAndKey kafkaCertKey = Ca.asCertAndKey(brokersSecret, cluster + "-kafka-0.key", cluster + "-kafka-0.crt");
-                        X509Certificate caCertCO = (X509Certificate) x509.generateCertificate(new ByteArrayInputStream(decoder.decode(clusterCaCertificateSecret.getData().get("ca.crt"))));
-                        X509Certificate caCertKafka = (X509Certificate) x509.generateCertificate(new ByteArrayInputStream(kafkaCertKey.cert()));
-
-                        KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
-                        kmf.init(setupKeyStore(clusterCaKeySecret, x509, password, caCertCO, kafkaCertKey), password);
-
-
-                        TrustManagerFactory tmf = TrustManagerFactory.getInstance("X509");
-                        tmf.init(setupTrustStore(caCertKafka, password, caCertCO));
-                        SSLContext ctx = SSLContext.getInstance("TLS");
-                        ctx.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
-
-                        factory = ctx.getSocketFactory();
-
-                        String host = pod.getStatus().getPodIP();
-                        int port = 2181;
-
-                        SSLSocket socket = (SSLSocket) factory.createSocket();
-                        log.debug("Connecting client to {}:{}", host, port);
-                        try {
-                            socket.connect(new InetSocketAddress(host, port), 10_000);
-                        } catch (ConnectException | SocketTimeoutException e) {
-                            log.error("Could not connect " + e.getMessage());
+                try {
+                    CertificateFactory x509 = CertificateFactory.getInstance("X.509");
+                    Base64.Decoder decoder = Base64.getDecoder();
+                    char[] password = "password".toCharArray();
+                    Pod pod0 = podOperations.get(namespace, name + "-" + 0);
+                    SecretOperator secretOperations = new SecretOperator(vertx, client);
+                    String cluster = pod0.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
```

Comment:
You can just pass the cluster name as a parameter. Getting it from a pod like this is a bit ugly.